src/SFA.DAS.AANHub.Data/AANDataContext.cs

```diff
@@ -1,136 +1,28 @@
-using SFA.DAS.AANHub.Data.Configuration;
-using SFA.DAS.AANHub.Domain.Configuration;
+﻿using Microsoft.EntityFrameworkCore;
 using SFA.DAS.AANHub.Domain.Entities;
 using SFA.DAS.AANHub.Domain.Interfaces;
 using System.Diagnostics.CodeAnalysis;
-using Calendar = SFA.DAS.AANHub.Domain.Entities.Calendar;
-
 
 namespace SFA.DAS.AANHub.Data
 {
     [ExcludeFromCodeCoverage]
-    public class AanDataContext : DbContext,
-        IRegionsContext,
-        IMembersContext,
-        IApprenticesContext,
-        IEmployersContext,
-        IPartnersContext,
-        IAdminsContext,
-        ICalendarsContext,
-        ICalendarPermissionsContext,
-        IMemberPermissionsContext
+    public class AanDataContext : DbContext, IAanDataContext
     {
-        private readonly ApplicationSettings? _configuration;
-
-        public virtual DbSet<Region> Regions { get; set; } = null!;
-        public virtual DbSet<Member> Members { get; set; } = null!;
-        public virtual DbSet<Apprentice> Apprentices { get; set; } = null!;
-        public virtual DbSet<Employer> Employers { get; set; } = null!;
-        public virtual DbSet<Partner> Partners { get; set; } = null!;
-        public virtual DbSet<Admin> Admins { get; set; } = null!;
-        public virtual DbSet<Calendar> Calendars { get; set; } = null!;
-        public virtual DbSet<CalendarPermission> CalendarPermissions { get; set; } = null!;
-        public virtual DbSet<MemberPermission> MemberPermissions { get; set; } = null!;
-
-        DbSet<Region> IEntityContext<Region>.Entities => Regions;
-        DbSet<Member> IEntityContext<Member>.Entities => Members;
-        DbSet<Apprentice> IEntityContext<Apprentice>.Entities => Apprentices;
-        DbSet<Employer> IEntityContext<Employer>.Entities => Employers;
-        DbSet<Partner> IEntityContext<Partner>.Entities => Partners;
-        DbSet<Admin> IEntityContext<Admin>.Entities => Admins;
-        DbSet<Calendar> IEntityContext<Calendar>.Entities => Calendars;
-        DbSet<CalendarPermission> IEntityContext<CalendarPermission>.Entities => CalendarPermissions;
-        DbSet<MemberPermission> IEntityContext<MemberPermission>.Entities => MemberPermissions;
-
-        public AanDataContext(DbContextOptions<AanDataContext> options) : base(options)
-        {
-        }
-
-        public AanDataContext(IOptions<ApplicationSettings> config, DbContextOptions<AanDataContext> options) : base(options)
-        {
-            _configuration = config.Value;
-        }
-
-        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
-        {
-            if (_configuration == null)
-            {
-                return;
-            }
-
-            var connection = new SqlConnection
-            {
-                ConnectionString = _configuration.DbConnectionString
-            };
-
-            optionsBuilder.UseSqlServer(connection);
-
-        }
+        public DbSet<Region> Regions { get; set; } = null!;
```

Comment:
There is a cleaner way of dong this as shown [here](https://learn.microsoft.com/en-us/ef/core/miscellaneous/nullable-reference-types#dbcontext-and-dbset)

Reply:
Now updated